crates/transaction-pool/src/maintain.rs

```diff
@@ -494,6 +495,82 @@ pub async fn maintain_transaction_pool<N, Client, P, St, Tasks>(
 
                 // keep track of mined blob transactions
                 blob_store_tracker.add_new_chain_blocks(&blocks);
+
+                // At Osaka transition, we need to convert blobs to new format
+                if !chain_spec.is_osaka_active_at_timestamp(tip.timestamp()) &&
+                    chain_spec.is_osaka_active_at_timestamp(tip.timestamp().saturating_add(12))
+                {
+                    let pool = pool.clone();
+                    let spawner = task_spawner.clone();
+                    let client = client.clone();
+                    task_spawner.spawn(Box::pin(async move {
+                        let mut interval = tokio::time::interval(Duration::from_secs(1));
+                        loop {
+                            // Loop and replace blob transactions until we reach Osaka transition
+                            // block after which no legacy blobs are going to be accepted.
+                            let last_iteration =
+                                client.latest_header().ok().flatten().is_none_or(|header| {
+                                    client
+                                        .chain_spec()
+                                        .is_osaka_active_at_timestamp(header.timestamp())
+                                });
+
+                            let AllPoolTransactions { pending, queued } = pool.all_transactions();
+                            for tx in pending.into_iter().chain(queued) {
+                                if !tx.transaction.is_eip4844() {
+                                    continue;
+                                }
+                                let tx_hash = *tx.transaction.hash();
+
+                                // Fetch sidecar from the pool
+                                let Ok(Some(sidecar)) = pool.get_blob(tx_hash) else {
+                                    continue;
+                                };
+                                // Ensure it is a legacy blob
+                                if !sidecar.is_eip4844() {
+                                    continue;
+                                }
+                                // Remove transaction and sidecar from the pool, both are in memory
+                                // now
+                                let Some(tx) = pool.remove_transactions(vec![tx_hash]).pop() else {
+                                    continue;
+                                };
+                                pool.delete_blob(tx_hash);
```

Comment:
there's a potential race condition here if block building is still on going, that's why I think we should sleep for 2.5s before we start the loop by that time we can assume that there's already 1 block ready and we can be a bit more reckless here